### PR TITLE
chore(docker): containerize repo

### PR DESCRIPTION
**Pull request by shortcuts**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | yes
| New feature?      | no
| BC breaks?        | no
| Related Issue     | Fixes https://github.com/algolia/algoliasearch-client-python/issues/517

## Describe your change

Add a Dockerfile and a guide to setup Docker for the repository

## What problem is this fixing?

Ease contributions to the Python client

## How to test?

- Follow [README](https://github.com/algolia/algoliasearch-client-python/blob/1bac766fd921910a3423afed77fc77b456ba77bd/DOCKER_README.md) steps
- In image's bash run 
  - `tox -e format` to run linter check
  - `tox -e types` to run type check
  - `tox -e py38-sync,py38-async` to run tests for Python 3.8

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Dockerfile
+FROM python:3.8.2
+
+RUN apt-get update -y \
+    && apt-get install -y --no-install-recommends \
+        wget \
+        zip \
+        unzip \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
```
Comment (DevinCodes) [inline Dockerfile:10]: Did you try building the image without this? IIRC this was specific to PHP, so I don't think we need it here.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+# install dev env dependencies
+RUN pip install --upgrade pip && \
+    pip install -r requirements.txt
+
+# setup dev env
+RUN python3 setup.py install
+
+WORKDIR /algoliasearch
+ADD . /algoliasearch/
```
Comment (DevinCodes) [inline Dockerfile:22]: I think this should work as well.
```suggestion
WORKDIR /algoliasearch
ADD . /algoliasearch/

# install dev env dependencies
RUN pip install --upgrade pip && \
    pip install -r requirements.txt

# setup dev env
RUN python3 setup.py install
```

**Review by DevinCodes — CHANGES_REQUESTED** [review]
Just left 2 comments, after which I think it's good to go! 🚀

File: Dockerfile (hunk 2 above)
Comment (shortcuts) [reply Dockerfile]: It does! https://github.com/algolia/algoliasearch-client-python/pull/538/commits/2259a8361ba07211128152fe7f5ffbb9c25c01c2

File: Dockerfile (hunk 1 above)
Comment (shortcuts) [reply Dockerfile]: Oops indeed it's not needed! https://github.com/algolia/algoliasearch-client-python/pull/538/commits/2259a8361ba07211128152fe7f5ffbb9c25c01c2

**Review by DevinCodes — APPROVED** [review]

**Pull request merged**